File: codes/day-4/AbstractDemo/DataAccessLibrary/DataAccessFactory.cs
namespace DataAccessLibrary
{
    public class DataAccessFactory
    {
        public static DataAccess CreateDataAccessObject(int choice)
        {
            DataAccess dataAccess;
            switch (choice)
            {
                case 1:
                    dataAccess = new DbDataAccess("db path");
                    break;

                case 2:
                    dataAccess = new XmlFileDataAccess("excel file path");
                    break;

                default:
                    dataAccess = null;
                    break;
            }
            return dataAccess;
        }
    }
}

File: codes/day-4/js-app/src/dataaccessfactory.ts
import { DataAccess } from "./dataaccess";
import { DbDataAccess } from "./dbdataaccess";

export class DataAccessFactory{
    static createDataAccessObject(choice: number):DataAccess|null {
        let dataAccess: DataAccess|null;
        switch (choice) {
            case 1:
                dataAccess = new DbDataAccess("db path");
                break;
        
            default:
                dataAccess = null;
                break;
        }
        return dataAccess
    }
}